src/lexitools/commands/correspondences.py

```diff
@@ -31,150 +33,195 @@ def register(parser):
-            help='select a display')
+        '--display',
+        action='store',
+        default=None,
+        help='select a display')
     parser.add_argument(
-            '--threshold',
-            action='store',
-            default=1,
-            type=int,
-            help='select a threshold')
+        '--threshold',
+        action='store',
+        default=1,
+        type=int,
+        help='select a threshold')
     parser.add_argument(
-            '--cutoff',
-            action='store',
-            default=2,
-            type=float,
-            help='select a threshold')
+        '--cutoff',
+        action='store',
+        default=2,
+        type=float,
+        help='select a threshold')
 
+    parser.add_argument(
+        '--model',
+        action='store',
+        default='bipa',
+        type=str,
+        help='select a sound class model: asjp, bipa, color, sca')
 
-def run(args):
+    parser.add_argument(
+        '--concepts',
+        action='store',
+        default='Holman-2008-40',
+        type=str,
+        help='select a concept list to filter on')
 
-    ds = get_dataset(args)
-    clts = args.clts.from_config().api
-    bipa = clts.transcriptionsystem_dict['bipa']
-    color = clts.soundclasses_dict['color']
-    asjp = clts.soundclasses_dict['asjp']
-    sca = clts.soundclasses_dict['sca']
+def iter_phonemes(tokens):
+    for segment in tokens:
+        if "/" in segment:
+            segment = segment.split("/")[1]
+        yield segment
 
 
-    columns=('parameter_id', 'concept_name', 'language_id', 'language_name',
-            'value', 'form', 'segments', 'language_glottocode',
-            'concept_concepticon_id', 'language_latitude',
-            'language_longitude')
-    namespace=(('concept_name', 'concept'), ('language_id', 'doculect'),
-            ('segments', 'tokens'), ('language_glottocode', 'glottolog'),
-            ('concept_concepticon_id', 'concepticon'), ('language_latitude',
-                'latitude'), ('language_longitude', 'longitude'), ('cognacy',
-                    'cognacy'), ('cogid_cognateset_id', 'cogid'))
+def sound_corresp_matrix(G, attr="total freq"):
+    alphabet = G.nodes()
+    matrix = nx.to_numpy_matrix(G, weight=attr)
+    freq = [[G.node[s][attr]] for s in alphabet]
+    return np.append(matrix, freq, axis=1).tolist()
 
 
-    lex = LexStat.from_cldf(ds.cldf_dir.joinpath('cldf-metadata.json'))
+def run(args):
+    ds = get_dataset(args)
+    clts = args.clts.from_config().api
+    if args.model == "bipa":
+        sound_class = clts.transcriptionsystem_dict[args.model]
+    else:
+        sound_class = clts.soundclasses_dict[args.model]
+
+    concepticon = Concepticon(Config.from_file().get_clone("concepticon"))
+    concept_dict = concepticon.conceptlists[args.concepts].concepts
+    concepts = {c.concepticon_id for c in concept_dict.values() if c.concepticon_id}
+
+    # columns=('parameter_id', 'concept_name', 'language_id', 'language_name',
```

Comment:
Note that we will need these later, and that it is essential for the rendering of certain aspects to adhere to the namespace. Leave it uncommented now, but basically, we need to be explicit about namespaces when importing cldf datasets to lingpy.

Reply:
Ok, it is unclear how these are useful, they were not and are not used in the code. I thought you had written them as a sort of documentation to inform me about the structure of the LexStat object,. I'll keep these commented but will not remove it.